src/Cake.Common.Tests/Unit/Tools/VSWhere/Product/VSWhereProductTests.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Cake.Common.Tests.Fixtures.Tools.VSWhere.Product;
+using Cake.Testing;
+using Cake.Testing.Xunit;
+using Xunit;
+
+namespace Cake.Common.Tests.Unit.Tools.VSWhere.Product
+{
+    public class VSWhereProductTests
+    {
+        public sealed class TheProductsMethod
+        {
+            [Fact]
+            public void Should_Throw_If_Settings_Are_Null()
+            {
+                // Given
+                var fixture = new VSWhereProductFixture();
+                fixture.Settings = null;
+
+                // When
+                var result = Record.Exception(() => fixture.Run());
+
+                // Then
+                Assert.IsArgumentNullException(result, "settings");
+            }
+
+            [Fact]
+            public void Should_Throw_If_VSWhere_Executable_Was_Not_Found()
+            {
+                // Given
+                var fixture = new VSWhereProductFixture();
+                fixture.GivenDefaultToolDoNotExist();
+
+                // When
+                var result = Record.Exception(() => fixture.Run());
+
+                // Then
+                Assert.IsCakeException(result, "VSWhere: Could not locate executable.");
+            }
+
+            [Theory]
+            [InlineData("/bin/vswhere/vswhere.exe", "/bin/vswhere/vswhere.exe")]
+            [InlineData("./tools/vswhere/vswhere.exe", "/Working/tools/vswhere/vswhere.exe")]
+            public void Should_Use_VSWhere_Executable_From_Tool_Path_If_Provided(string toolPath, string expected)
+            {
+                // Given
+                var fixture = new VSWhereProductFixture();
+                fixture.Settings.ToolPath = toolPath;
+                fixture.GivenSettingsToolPathExist();
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal(expected, result.Path.FullPath);
+            }
+
+            [WindowsTheory]
+            [InlineData("C:/vswhere/vswhere.exe", "C:/vswhere/vswhere.exe")]
```

Comment:
@patriksvensson That throws and error saying inline data is not allowed for a Fact.  I am more than happy to make the change and just put it directly in the test.  It is like that in other parts of the code, an example would be the nuget tests, so that is why I did it that way.